api/volume.go

```diff
@@ -204,7 +212,7 @@ func (s *Server) VolumeAttach(rw http.ResponseWriter, req *http.Request) error {
 	id := mux.Vars(req)["name"]
 
 	obj, err := util.RetryOnConflictCause(func() (interface{}, error) {
-		return s.m.Attach(id, input.HostID, input.DisableFrontend, input.AttachedBy)
+		return s.m.Attach(id, input.HostID, input.DisableFrontend, input.AttachedBy, input.AttacherType, input.AttachmentID)
```

Comment:
I'm thinking if `input.AttachmentID` can be created in `manager.Attach` rather than an input.

Reply:
The client will need to remember the AttachmentID so that it can delete the ticket later on. Could you give a suggestion about how we would implement this behavior if input.AttachmentID can be created in manager.Attach rather than an input?

Reply:
Sorry... Can you elaborate more about `the client will need to remember the AttachmentID`? Where will Longhorn cache Attachments?